Report subset of config values in SHOW DIAGNOSTICS

mark-rushakoff opened the pull request:
Fixes #7821.

- [x] Rebased/mergable
- [x] Tests pass
- [ ] CHANGELOG updated

e-dard commented on cmd/influxd/run/config.go:
```diff
@@ -306,3 +307,60 @@ func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value, structKey
+func (c *Config) diagnosticsClients() map[string]diagnostics.Client {
+	// Config settings that are always present.
+	m := map[string]diagnostics.Client{
+		"config": c,
```
If we're including `c` do we need to explicitly include all the other things? I'm a bit unclear on this.

mark-rushakoff replied:
`c` implements `diagnostics.Client` directly with its own two, top-level config values `reporting-disabled` and `bind-address`, distinct from the other config sections.  The presence of the top-level / unscoped config settings at all is what makes this a little awkward.

Each key-value pair in the map is registered as a diagnostics client. This results in output like:

```
> show diagnostics
name: build
Branch  Build Time Commit  Version
------  ---------- ------  -------
unknown            unknown unknown


name: config
bind-address reporting-disabled
------------ ------------------
:8088        false


name: config-coordinator
log-queries-after max-concurrent-queries max-select-buckets max-select-point max-select-series query-timeout write-timeout
----------------- ---------------------- ------------------ ---------------- ----------------- ------------- -------------
0s                0                      0                  0                0                 0s            10s


name: config-cqs
enabled run-interval
------- ------------
true    1s

...
```

e-dard commented on services/collectd/config.go:
```diff
@@ -128,3 +129,37 @@ func (c *Config) Validate() error {
+		if !cc.Enabled {
+			r := make([]interface{}, 1, len(d.Columns))
+			r[0] = false
+			d.AddRow(r)
```
You could shorten this somewhat to just:

```go
d.AddRow([]interface{}{false})
```

mark-rushakoff replied:
I don't recall why I thought the capacity of the row had to match the length of the header row. Changing this and the other occurrences.

e-dard commented on services/graphite/config.go:
```diff
@@ -253,3 +254,37 @@ func (c *Config) validateTag(keyValue string) error {
+		if !cc.Enabled {
+			r := make([]interface{}, 1, len(d.Columns))
+			r[0] = false
+			d.AddRow(r)
```
Same as above

mark-rushakoff commented:
Rebased and removed commit message that noted the config information was included in /debug/vars output. I think that may have been true when the PR was originally opened but that is no longer the case after rebasing. I think it's fine if this is only available via SHOW DIAGNOSTICS.

mark-rushakoff commented:
Not merging this yet: before this change, `TestShard_WritePoints_FieldConflictConcurrentQuery` completes in about 3s on my machine; after, it takes about 99s! It doesn't look like this change should have affected this test but I'm going to continue digging into it.

mark-rushakoff commented:
Turns out the timing of that test was already sporadic on master as of fe5a5f38d7fafaa08260c30a5edaab571af5e909, prior to this commit, so I'm merging this PR. I'll still try and bisect to find where the test started getting inconsistent timing, but that's out of scope from the PR now.

```
$ time ./tsdb.test -test.run=ConcurrentQuery -test.cpuprofile /tmp/cq.cpuprof
PASS

real	0m0.486s
user	0m0.795s
sys	0m0.103s

$ time ./tsdb.test -test.run=ConcurrentQuery -test.cpuprofile /tmp/cq.cpuprof
PASS

real	1m20.424s
user	0m1.183s
sys	0m0.455s
```
